gt/lua_lpm.c

```diff
@@ -192,8 +194,11 @@ l_lpm_lookup(lua_State *l)
 		luaL_error(l, "Expected two arguments, however it got %d arguments",
 			lua_gettop(l));
 
-	lua_pushinteger(l, lpm_lookup_ipv4(lpm, ip));
+	ret = lpm_lookup_ipv4(lpm, ip);
+	if (ret == -ENOENT)
+		luaL_error(l, "Could not find IPv4 entry while doing Lua LPM lookup");
 
+	lua_pushinteger(l, ret);
```

Comment:
Raising an exception isn't a very cheap operation, and this function is called at least once for each packet in the production policy. Moreover, the production policy is already expecting a negative return, but not an exception. Thus, I think we should avoid this change. What do you think?

This comment also applies to the IPv6 case.

Reply:
Sure, let's let the policy handle it.